spec/controllers/reviews_controller_spec.rb

```diff
@@ -0,0 +1,162 @@
+    end
+
+    it "updates the restaurant's rating" do
+      post :create, params: { restaurant_id: rest.id,
+                              reviewer_name: "Dotan Langsam Schwartz",
+                              rating: 1,
+                              comments: 'I like it here'}
+      expect(response.status).to eq(200)
+      expect(Restaurant.find(rest.id).rating).to eq(1)
+
+      post :create, params: { restaurant_id: rest.id,
+                              reviewer_name: "Pierre Schwartz",
+                              rating: 3,
+                              comments: 'I like it here too'}
+      expect(response.status).to eq(200)
+      expect(Restaurant.find(rest.id).rating).to eq 2
+    end
+
+  end
+
+  describe '#update' do
+    let(:rest) { FactoryBot.create(:restaurant) }
+    let(:review) { FactoryBot.create(:review, restaurant_id: rest.id) }
+    it "updates the comment name" do
+      patch :update, params: {id: review.id, comments: 'This is an updated comment' }
+      body = JSON.parse(response.body)
+      expect(response.status).to eq 200
+      expect(body['comments']).to eq('This is an updated comment')
+    end
+
+    it "fails to update a non existent id" do
+      expect {
+        patch :update, params: { id: review.id + 1, comments: 'This is an updated comment' }
+      }.to raise_error(ActiveRecord::RecordNotFound)
+    end
+
+    it "updates the restaurant's rating when the rating in the review changes" do
+      post :create, params: { restaurant_id: rest.id, reviewer_name: "Dotan Langsam Schwartz", rating: 1}
+      expect(response.status).to eq(200)
+
+      post :create, params: { restaurant_id: rest.id, reviewer_name: "Pierre Schwartz", rating: 3 }
+
+      review_id = JSON.parse(response.body)['id']
+      expect(response.status).to eq(200)
+
+      patch :update, params: {id: review_id, rating: 3 }
+      expect(response.status).to eq(200)
+      new_rating = Review.where(restaurant_id: rest.id).average(:rating).round(0)
+      expect(Restaurant.find(rest.id).rating).to eq(new_rating)
+    end
+  end
+
+  describe '#index' do
+    let!(:rest1) { FactoryBot.create(:restaurant, name: 'rest1') }
+    let!(:review1) { FactoryBot.create(:review, restaurant_id: rest1.id) }
+
+    let!(:rest2) { FactoryBot.create(:restaurant, name: 'rest2') }
+    let!(:review2) { FactoryBot.create(:review, restaurant_id: rest2.id) }
+
+    it "retrieve all reviews" do
+      get :index
+      body = JSON.parse(response.body)
+      expect(response.status).to eq 200
+      expect(body.size).to eq 2
+    end
+  end
+
+  describe '#destroy' do
+    let!(:rest1) { FactoryBot.create(:restaurant, name: 'rest1') }
+    let!(:review1) { FactoryBot.create(:review, restaurant_id: rest1.id) }
+
+    let!(:rest2) { FactoryBot.create(:restaurant, name: 'rest2') }
+    let!(:review2) { FactoryBot.create(:review, restaurant_id: rest2.id) }
+
+    it "deletes the review by its id" do
+      get :index
```

Comment:
Why do you need this first part for testing update? Don't you trust FactoryBot to create 2 restaurants?